frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -1406,6 +1407,14 @@ const IR::Node* TypeInference::postorder(IR::Parameter* param) {
         return param;
     BUG_CHECK(!paramType->is<IR::Type_Type>(), "%1%: unexpected type", paramType);
 
+    if (param->defaultValue != nullptr) {
+        auto init = assignment(param, paramType, param->defaultValue);
+        if (init == nullptr)
+            return param;
+        if (param->defaultValue != init)
+            param->defaultValue = init;
```

Comment:
Are you overloading `defaultValue`? Or is assignment likely to return a simplified expression?

Reply:
Assignment does type inference, and may return a new expression with modified types. In particular, `int` types may be converted to `bit<>` types.